lib/ansible/modules/windows/win_say.ps1

```diff
@@ -19,59 +19,57 @@
 # WANT_JSON
 # POWERSHELL_COMMON
 
-$params = Parse-Args $args;
-$result = New-Object PSObject;
-$msg = Get-AnsibleParam -obj $params -name "msg"
+$params = Parse-Args $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+
+$msg = Get-AnsibleParam -obj $params -name "msg" -type "str"
 $msg_file = Get-AnsibleParam -obj $params -name "msg_file" -type "path"
 $start_sound_path = Get-AnsibleParam -obj $params -name "start_sound_path" -type "path"
 $end_sound_path = Get-AnsibleParam -obj $params -name "end_sound_path" -type "path"
-$voice = Get-AnsibleParam -obj $params -name "voice"
-$speech_speed = Get-AnsibleParam -obj $params -name "speech_speed"
+$voice = Get-AnsibleParam -obj $params -name "voice" -type "str"
+$speech_speed = Get-AnsibleParam -obj $params -name "speech_speed" -type "str"
+
+$result = @{ }
+
 $speed = 0
 $words = $null
 
 if ($speech_speed -ne $null) {
    try {
-     $speed = [convert]::ToInt32($speech_speed, 10)
+      $speed = [convert]::ToInt32($speech_speed, 10)
    } catch {
-       Fail-Json $result "speech_speed needs to a integer in the range -10 to 10.  The value $speech_speed could not be converted to an integer."
-
+      Fail-Json $result "speech_speed needs to a integer in the range -10 to 10.  The value $speech_speed could not be converted to an integer."
    }
    if ($speed -lt -10 -or $speed -gt 10) {
-       Fail-Json $result "speech_speed needs to a integer in the range -10 to 10.  The value $speech_speed is outside this range."
+      Fail-Json $result "speech_speed needs to a integer in the range -10 to 10.  The value $speech_speed is outside this range."
    }
 }
 
-
 if ($msg_file -ne $null -and $msg -ne $null ) {
    Fail-Json $result "Please specify either msg_file or msg parameters, not both"
 }
 
-if ($msg_file -eq $null -and $msg -eq $null -and $start_sound_path -eq $null -and $end_sound_path -eq $null) {
```

Comment:
RiIght, probably got squeeshed when I was investigating how to make mutual-exclusive and required options like we do in Python. What I was doing was needlessly complex so I dropped it. Still I think we need such functionality. I am sure @nitzmahone has some clear ideas about this too :-)